File: Solutions/3407/c++/program.cpp
// Copy from Quera

#include <iostream>
using namespace std;

int a[100][100];

int main()
{
    int n,m,k,x,y;
    cin>>n>>m>>k;
    for(int i=0;i<k;i++)
    {
            cin>>x>>y;
            a[x-1][y-1]=-100;
            if(y-2<m && y-2>-1 && x-1<n && x-1>-1)a[x-1][y-2]++;
            if(y<m && y>-1 && x-1<n && x-1>-1)a[x-1][y]++;
            if(y-1<m && y-1>-1 && x-2<n && x-2>-1)a[x-2][y-1]++;
            if(y-1<m && y-1>-1 && x<n && x>-1)a[x][y-1]++;
            if(y<m && y>-1 && x<n && x>-1)a[x][y]++;
            if(y-2<m && y-2>-1 && x<n && x>-1)a[x][y-2]++;
            if(y<m && y>-1 && x-2<n && x-2>-1)a[x-2][y]++;
            if(y-2<m && y-2>-1 && x-2<n && x-2>-1)a[x-2][y-2]++;
    }
    for(int i=0;i<n;i++)
    {
        for(int j=0;j<m;j++)
        {
            if(a[i][j]<0)cout<<"* ";
            else cout<<a[i][j]<<" ";
        }
        cout<<endl;
    }
    return 0;
}

File: Solutions/3407/java/program.java
// Copy from Quera

import java.util.Scanner;

public class Main {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int n = scanner.nextInt();
		int m = scanner.nextInt();
		int k = scanner.nextInt();
		int i,j;
		String[][] a = new String[n][m];
		int[][] b = new int[n][m];
		for (int f = 0;f < k; f++) {
			i = scanner.nextInt();
			j = scanner.nextInt();
			a[i-1][j-1]="*";
			if(((i-2)>=0)&&((i-2)<n)&&((j-2)>=0)&&((j-2)<m)) b[i-2][j-2]++;
			if(((i-2)>=0)&&((i-2)<n)&&((j-1)>=0)&&((j-1)<m)) b[i-2][j-1]++;
			if(((i-2)>=0)&&((i-2)<n)&&((j)>=0)&&((j)<m)) b[i-2][j]++;
			if(((i-1)>=0)&&((i-1)<n)&&((j-2)>=0)&&((j-2)<m)) b[i-1][j-2]++;
			if(((i-1)>=0)&&((i-1)<n)&&((j)>=0)&&((j)<m)) b[i-1][j]++;
			if(((i)>=0)&&((i)<n)&&((j-2)>=0)&&((j-2)<m)) b[i][j-2]++;
			if(((i)>=0)&&((i)<n)&&((j-1)>=0)&&((j-1)<m)) b[i][j-1]++;
			if(((i)>=0)&&((i)<n)&&((j)>=0)&&((j)<m)) b[i][j]++;

		}
		for (int l = 0; l < n; l++) {
			for (int l2 = 0; l2 < m; l2++) {
				if("*".equals(a[l][l2]))
					System.out.print("* ");
				else
					System.out.print(b[l][l2]+" ");
			}
			System.out.println();
		}

		scanner.close();
	}

}